### ISSUE TITLE
Macro for registering types for syncing

### ISSUE THREAD

**Issue by randomPoison** [randomPoison/amethyst-editor-sync#19]
Originally requested by @jojolepro in randomPoison/amethyst-editor#28.

A macro for registering components/resources for synchronization would be helpful in cutting down boilerplate when registering a large number of types. Such a macro should define a way to register both components and resources, and should only require that the type be specified for each. It should automatically generate the name by stringifying the type name used for registration (the [`stringify`](https://doc.rust-lang.org/beta/std/macro.stringify.html) macro can be used for this). It should generate code that just calls `sync_component` and `sync_resource`, returning a new `SyncEditorBundle`.

A strawman syntax could be:

```
let editor_bundle = sync_editor! {
    components:
        Foo,
        Bar,
        Baz,

    resources:
        Abba,
        Baab,
        Flub,
};
```

**Comment by jojolepro** [comment]
Other syntax that would be easier to implement:
```rs
let editor_bundle = sync_editor! {
    [Foo, Bar, Baz],
    [Abba, Baab, 
        Flub],
}
```

**Comment by randomPoison** [comment]
This was implemented in #21.
